tests/paz/pipelines/pix2pose_metric_test.py

```diff
@@ -0,0 +1,167 @@
+         0.3050982069800776, -0.5951250307633585,
+         0.743466715795677, -0.28761117420744436, 0.6866522290644125,
+         0.6676740178175343]).reshape(3, 3)
+    translation = np.array([114.54312464515644, 71.49346046440272, 955.3250389603148]).reshape(3,1)                                                                       
+    return rotation, translation
+
+
+@pytest.fixture
+def pose_C(pose_A):
+    rotation, translation= pose_A[0], pose_A[1]
+    rotation_x=build_rotation_matrix_x(20)
+    transformed_rotation=rotation_x @ rotation                                                                      
+    return transformed_rotation, translation
+
+
+@pytest.fixture
+def pix2pose_pose6d(image_with_powerdrill):
+    cv2.ocl.setUseOpenCL(False)
+    cv2.setNumThreads(1)
+    cv2.setRNGSeed(777)
+    camera = Camera()
+    camera.intrinsics_from_HFOV(55, (480, 640, 3))
+    detector = PIX2POSEPowerDrill(camera, score_thresh=0.45, offsets=[0.25, 0.25], epsilon=0.015)
+    inferences = detector(image_with_powerdrill)
+    pose6D = inferences['poses6D'][0]
+    return pose6D
+
+
+@pytest.fixture
+def points3D():
+    X=np.linspace(1e-1,1e-3,  num=9128).reshape(9128,1)
+    Y=np.linspace(1e-2,1e-4,  num=9128).reshape(9128,1)
+    Z=np.linspace(1e-4,1e-6,  num=9128).reshape(9128,1)
+    points3D=np.hstack((X,Y,Z))
+    return points3D
+
+
+def test_quaternion_translation_shape(pix2pose_pose6d):
+    pred_quaternion, pred_translation = pix2pose_pose6d.quaternion, pix2pose_pose6d.translation
+    pred_pose = Pose6D(pred_quaternion, pred_translation)
+    pred_quaternion = get_quaternion(pred_pose)
+    assert (pred_quaternion.shape[0] == 4)
+    assert None not in pred_quaternion
+    pred_translation = get_translation(pred_pose)
+    assert (pred_translation.shape[0] == 3)
+    assert None not in pred_translation
+
+
+def test_transform_points3D(points3D, pose_A):
+    real_quaternion, real_translation = rotation_matrix_to_quaternion(
+        pose_A[0]), pose_A[1]
+    real_rotation = quaternion_to_rotation_matrix(real_quaternion)
+    assert (points3D.shape[1] == 3)
+    assert None not in points3D
+    transformed_points = transform_points3D(points3D, real_rotation, real_translation)
+    assert (transformed_points.shape[1] == 3)
+    assert None not in transformed_points
+
+
+def test_ADI_zero(pose_A, pose_B, pose_C, points3D):
+    a_quaternion, a_translation = rotation_matrix_to_quaternion(pose_A[0]), pose_A[1]
+    b_quaternion, b_translation = rotation_matrix_to_quaternion(pose_B[0]), pose_B[1]
+    c_quaternion, c_translation = rotation_matrix_to_quaternion(pose_C[0]), pose_C[1]     
+    b_pose = Pose6D(b_quaternion, b_translation)
+    a_pose = Pose6D(a_quaternion, a_translation)
+    c_pose = Pose6D(c_quaternion, c_translation)
+    adi_a = compute_ADI(a_pose, a_pose, points3D)
+    adi_b = compute_ADI(b_pose, b_pose, points3D)
+    add_c = compute_ADD(c_pose, c_pose, points3D)
```

Comment:
We should use capitalize these variable names for example as: ADI_A or ADD_A. This since ADI and ADI are acronyms and A capital since is the convention we took for the pose_A.

Reply:
resolved